### PR TITLE
Add success/fail summary files, primer_val summary file, fix some bugs

### PR DESCRIPTION
**Pull request by chrisAta**
This PR merges changes that achieve:
- New summary .tsv file for runs that fail sanity checking or QC
- New summary .tsv file for runs that succeed (with details about which have ASV results and which don't)
- New summary .json file for primer validation
- Publishes SeqFu and suffixheadercheck output files to qc dir
- Some bug fixes and other small updates

### CODE REVIEW

File: subworkflows/local/mapseq_asv_krona_swf.nf [hunk 1]
```diff
@@ -70,7 +70,8 @@ workflow MAPSEQ_ASV_KRONA {
                                       }
         
         MAKE_ASV_COUNT_TABLES(
-            final_asv_count_table_input
+            final_asv_count_table_input,
+            krona_tuple[4]
```
Comment (mberacochea) [inline subworkflows/local/mapseq_asv_krona_swf.nf:74]: shouldn't this be korna_tuple -> it[4]?

Comment (chrisAta) [reply subworkflows/local/mapseq_asv_krona_swf.nf:74]: this isn't needed, krona_tuple is a 'constant' tuple defined for example here:

 https://github.com/EBI-Metagenomics/amplicon-pipeline/blob/1ce86a29a8c94b66300ae4e429e4d3e35adf4e04/workflows/pipeline.nf#L109

File: workflows/pipeline.nf [hunk 2]
```diff
@@ -284,4 +362,101 @@ workflow AMPLICON_PIPELINE {
...
+                    ]
+
+                if( primer_name.contains("_auto" )){
+                    new_primer["identification_strategy"] = "auto"
+                }
+                else{
+                    new_primer["identification_strategy"] = "std"
+                }
+
+                json_map["primers"].add(new_primer)
```
[the remark this answers is not in the document]
Comment (mberacochea) [reply workflows/pipeline.nf:448]: BTW, I haven't tested this.

**Review by KateSakharova — APPROVED** [review]
I'm checking your last results and it all looks great! I love it!
 I'm wondering is is hard to move your examples with qc_not_passed, no_asv, seqfu_fail to a test here? It might be a test with only failed runs by different reasons, no need to add successful test, just to show various failures

File: workflows/pipeline.nf [hunk 3]
```diff
@@ -244,6 +276,52 @@ workflow AMPLICON_PIPELINE {
...
+    are consistent i.e. ASVs in read count files, ASV sequences in FASTA files, etc.
+    */
+    extract_asv_read_counts_input = MAPSEQ_ASV_KRONA_SILVA.out.asvs_left
+        .join(MAPSEQ_ASV_KRONA_PR2.out.asvs_left)
+
+    EXTRACT_ASV_READ_COUNTS(extract_asv_read_counts_input)
+    ch_versions = ch_versions.mix(EXTRACT_ASV_READ_COUNTS.out.versions)
+
+    extract_asvs_input = EXTRACT_ASV_READ_COUNTS.out.asvs_left
+                    .map{ meta, asvs_left ->
```
[the remark this answers is not in the document]
Comment (chrisAta) [reply workflows/pipeline.nf:291]: Ah yes.. I'm gonna do a pass of all the subworkflows too and replace anything like this with a filter first, thanks!

File: workflows/pipeline.nf (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisAta) [reply workflows/pipeline.nf:448]: Love this! Will change it and test it

**Comment by chrisAta** [comment]
> I'm checking your last results and it all looks great! I love it! I'm wondering is is hard to move your examples with qc_not_passed, no_asv, seqfu_fail to a test here? It might be a test with only failed runs by different reasons, no need to add successful test, just to show various failures

Good idea, I have a samplesheet of those already so should be easy to do, thank you!!

**Comment by chrisAta** [comment]
All changes done!

**Review by mberacochea — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**